docs/se/guides/migration_4x.adoc

```diff
@@ -0,0 +1,216 @@
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Helidon SE 4.x Upgrade Guide
+:description: Helidon SE 4.x Upgrade Guide
+:keywords: helidon, porting, migration, upgrade, incompatibilities
+:rootdir: {docdir}/../..
+
+In Helidon 4.x we have made some major changes to Helidon. Reactive engine has been removed. APIS and implementations rewritten in "blocking" paradigm. This guide will help you upgrade a Helidon MP 3.x application to 4.x.
+
+
+== Java 21 Runtime
+
+Java 17 is no longer supported in Helidon 4. Java 21 or newer is required. Please follow the instructions in xref:{rootdir}/about/prerequisites.adoc[Prerequisites] for proper installation.
+
+Helidon 4 no more uses Netty. Helidon SE is now running on Helidon WebServer which is based on Virtual threads technology, available in Java 21.
+
+
+== Deprecations
+
+Helidon Reactive APIs are now removed from Helidon 4. Helidon Reactive Engine is completely removed. Objects `Single` and `Multi`, as well as all reactive operators are no longer available. All modules of Helidon SE have been redesigned and rewritten to use "Blocking" paradigm (coding style).
+
+There is no more backward compatibility with Helidon SE version 3 and earlier! All reactive code written using Reactive paradigm should be redesigned and rewritten in blocking paradigm.
+
+== Server initialization and start up
+
+In Helidon 1.x-3.x to star a server the following actions should be done:
+
+[source, java]
+.Start Helidon SE 3.x Server
+----
+static Single<WebServer> startServer() {
+
+    Config config = Config.create();
+
+    WebServer server = WebServer.builder(createRouting(config))
+            .config(config.get("server"))
+            .addMediaSupport(JsonpSupport.create())
+            .build();
+
+    Single<WebServer> webserver = server.start();   <1>
+
+    webserver.thenAccept(ws -> {    <2>
+                System.out.println("WEB server is up! http://localhost:" + ws.port() + "/greet");
+                ws.whenShutdown().thenRun(() -> System.out.println("WEB server is DOWN. Good bye!"));
+            })
+            .exceptionallyAccept(t -> { <3>
+                System.err.println("Startup failed: " + t.getMessage());
+                t.printStackTrace(System.err);
+            });
+
+    return webserver;
+}
+----
+<1> Server is started in asynchronous way. A `Single` object is returned.
+<2> Wait for the Server to start and print the message in asynchronous way.
+<3> Gracefully handle exceptions if they occur during the initialization process.
+
+Since Helidon SE is reactive, during the start a `Single` object is returned, the server has been started in asynchronous way. We have to use reactive methods like `thenAccept` to wait for the server to start and the to perform the desired action. The exception handling should also be done in reactive way using the corresponding method.
+
+In Helidon there are no more Reactive APIs, so the Server is started much simpler:
```

Comment:
"In Helidon 4 there are..."
"...Server startup is much simpler..."

Reply:
ok :)